### PR TITLE
refactor rows in Files into its own component

### PR DESCRIPTION
**Pull request by songgao**
This is is purely a refactor PR, and *shouldn't* change any behaviors. This would be used in the move/copy modal dialog for Files. Since it moves stuff around, I was hoping to get this in first by its own to hopefully avoid conflicts.

### CODE REVIEW

**Comment by songgao** [comment]
bump!

**Comment by songgao** [comment]
⪮

File: shared/fs/index.js [hunk 1]
```diff
@@ -2,156 +2,89 @@
...
-        return (
-          <WrapRow key="">
-            <Text type="BodySmallError">This should not happen.</Text>
-          </WrapRow>
-        )
-    }
-  }
-
   render() {
+    console.log({songgao: 'Files', path: this.props.path, props: this.props})
```
Comment (chrisnojima) [inline shared/fs/index.js]: 🔥

File: shared/fs/index.js [hunk 2]
```diff
@@ -2,156 +2,89 @@
...
+          <Kb.Box style={styles.emptyContainer}>
+            <Kb.Text type="BodySmall">This is an empty folder.</Kb.Text>
+          </Kb.Box>
+        }
+      />
     )
     return (
-      <Box style={styleOuterContainer}>
-        <Box style={stylesContainer}>
+      <Kb.Box style={styles.outerContainer}>
```
Comment (chrisnojima) [inline shared/fs/index.js]: why not box2

File: shared/fs/row/rows-container.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+const mergeProps = (s, d, o: OwnProps) => ({
+  items: getItemsFromStateProps(s, o.path, o.sortSetting),
+  routePath: o.routePath,
+  ifEmpty: o.ifEmpty,
+})
+
+export default compose(
+  FilesLoadingHoc,
+  // $FlowIssue @jzila/@songgao lots of exposed flow issues here
```
Comment (chrisnojima) [inline shared/fs/row/rows-container.js]: this should get looked at at some point (not in this pr)

File: shared/fs/row/rows.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    )
+  }
+}
+
+const rowHeight = Styles.isMobile ? 64 : 40
+
+const styles = Styles.styleSheetCreate({
+  rowContainer: {
+    ...Styles.globalStyles.flexBoxColumn,
+    height: rowHeight,
```
Comment (chrisnojima) [inline shared/fs/row/rows.js]: you should be able to do flexShrink: 0

File: shared/fs/index.js (hunk 1 above)
Comment (songgao) [reply shared/fs/index.js]: This is the fire that warmed up the code freeze 😎

File: shared/fs/row/rows.js (hunk 4 above)
Comment (songgao) [reply shared/fs/row/rows.js]: Could you explain what `flexShrink` does here? I think the height was added for mobile for stable height; I don't remember for sure though.

Comment (chrisnojima) [reply shared/fs/row/rows.js]: if you set a height and flexShrink: 0 it shouldn't reduce the height (get rid of minHeight/maxHeight)

**Comment by songgao** [comment]
@chrisnojima ready for another 👀

File: shared/fs/index.js (hunk 2 above)
Comment (chrisnojima) [reply shared/fs/index.js]: you should switch to box2 for all these and get rid of most of these styles

File: shared/fs/index.js [hunk 5]
```diff
@@ -43,7 +42,7 @@ class Files extends React.PureComponent<FolderProps> {
     )
     return (
       <Kb.Box style={styles.outerContainer}>
-        <Kb.Box style={styles.container}>
+        <Kb.Box2 direction="vertical" fullHeight={true}>
```
Comment (chrisnojima) [inline shared/fs/index.js]: you can remove the container style

**Comment by songgao** [comment]
@chrisnojima ping! 🏓

**Pull request merged**